baseplate/lib/metrics.py

```diff
@@ -501,8 +505,13 @@ def metrics_client_from_config(raw_config: config.RawConfig) -> Client:
     """
     cfg = config.parse_config(
         raw_config,
-        {"metrics": {"namespace": config.String, "endpoint": config.Optional(config.Endpoint)}},
+        {
+            "metrics": {"namespace": config.String, "endpoint": config.Optional(config.Endpoint)},
```

Comment:
Rather than forcing the metrics client to know about span sampling, which is only a concern of the observer, I think it'd make more sense to have the observer do its own config parsing. That could look like adding a new classmethod to `MetricsBaseplateObserver` like so:

```python
class MetricsBaseplateObserver(BaseplateObserver):
    @classmethod
    def from_config_and_client(cls, raw_config: config.RawConfig, client: metrics.Client) -> "MetricsBaseplateObserver":
        cfg = config.parse_config(raw_config, {
            "metrics_observer": {
               "sample_rate": config.Optional(config.Float, default=1.0),
            }
        })

       return cls(client, sample_rate=cfg.metrics_observer.sample_rate)

    def __init__(self, client: metrics.Client, sample_rate: float):
        self.client = client
        self.sample_rate = sample_rate
```

Then you can update `Baseplate.configure_metrics` to use that factory function rather than directly calling the constructor:

```python
def configure_metrics(self, metrics_client: metrics.Client) -> None:
    from baseplate.observers.metrics import MetricsBaseplateObserver

    self._metrics_client = metrics_client
    observer = MetricsBaseplateObserver.from_client_and_config(metrics_client, self._app_config or {})
    self.register(observer)
```

Since that relies on `self._app_config` being present, this setting will only be available to clients using the modern `Baseplate(app_config)` setup, which should probably be documented.